src/errors.c

```diff
@@ -133,8 +133,14 @@ void Error_InvalidFilterPlacement(rax *entitiesRax) {
 }
 
 void Error_SITypeMismatch(SIValue received, SIType expected) {
-	ErrorCtx_SetError("Type mismatch: expected %s but was %s", SIType_ToString(expected),
+	char *buf = rm_malloc(256 * sizeof(char));
```

Comment:
If we believe `buf` can't grow beyond 1024 bytes let's allocate it on the stack, and modify `SIType_ToMultipleTypeString` to not perform any reallocations, probably returning early if the buffer too small (which shouldn't happen) maybe `ASSERT` in such case.

Reply:
ASSERT() was added and reallocation was removed from ```SIType_ToMultipleTypeString ```
[44e274c](https://github.com/RedisGraph/RedisGraph/pull/2691/commits/44e274c4d072d7f04252ce0831d735fcf35d53d2)